geomesa-geojson/geomesa-geojson-api/src/main/scala/org/locationtech/geomesa/geojson/query/GeoJsonQuery.scala

```diff
@@ -99,55 +120,64 @@ object GeoJsonQuery {
           Or(ors: _*)
 
         case (prop, v: JValue) =>
-          Equals(JsonPathParser.parse(prop), v.values)
+          Equals(checkJsonPaths(prop), v.values)
+      }
+      if (predicates.length > 1) {
+        And(predicates: _*)
+      } else {
+        predicates.head
       }
-      if (predicates.length > 1) { And(predicates: _*) } else { predicates.head }
     }
   }
 
   /**
     * Evaluate a complex predicate
     *
-    * @param path path that the predicate operates on
+    * @param prop property that the predicate operates on
     * @param json predicate json object
     * @return
     */
-  private def evaluatePredicate(path: Seq[PathElement], json: JObject): GeoJsonQuery = {
+  private def evaluatePredicate(prop: String, json: JObject): GeoJsonQuery = {
     json.obj.headOption match {
       case Some(("$bbox", v: JArray)) =>
         // { "$bbox" : [-180, -90, 180, 90] }
         val List(xmin, ymin, xmax, ymax) = v.values.asInstanceOf[List[Number]]
-        Bbox(path, xmin.doubleValue, ymin.doubleValue, xmax.doubleValue, ymax.doubleValue)
+        Bbox(prop, xmin.doubleValue, ymin.doubleValue, xmax.doubleValue, ymax.doubleValue)
 
       case Some(("$intersects", v: JObject)) =>
         // { "$intersects" : { "$geometry" : { "type" : "Point", "coordinates" : [30, 10] } } }
-        Intersects(path, evaluateGeometry(v))
+        Intersects(prop, evaluateGeometry(v))
 
       case Some(("$within", v: JObject)) =>
         // { "$within" : { "$geometry" : { "type" : "Polygon", "coordinates": [ [ [0,0], [3,6], [6,1], [0,0] ] ] } } }
-        Within(path, evaluateGeometry(v))
+        Within(prop, evaluateGeometry(v))
+
+      case Some(("$dwithin", v: JObject)) =>
+        // { "$dwithin" : { "$geometry" : { "type" : "Point", "coordinates" : [30, 10] }, "$dist" : 100.50, "$unit" : "feet" } }
```

Comment:
Oh, I see. Perhaps that's why DWithin was not included in the original implementation. We needed to query using DWithin for our purposes and didn't want to have a separate code path. Maybe this can be fixed with proper documentation, or this could be something we pull into Blyncsy's forked repo.